backend/src/main/java/com/gleam/backend/user/application/UserService.java

```diff
@@ -0,0 +1,25 @@
+package com.gleam.backend.user.application;
+
+import com.gleam.backend.user.domain.User;
+import com.gleam.backend.user.infrastructure.UserRepository;
+import com.gleam.backend.user.ui.dto.RegisterUserRequest;
+import org.springframework.stereotype.Service;
+
+@Service
+public class UserService {
+    private final UserRepository userRepository;
+
+    public UserService(UserRepository userRepository) {
+        this.userRepository = userRepository;
+    }
+
+    public User getOrCreateUser(RegisterUserRequest requestDto) {
+        var optionalUser = userRepository.findByUsername(requestDto.username());
+        if (optionalUser.isEmpty()) {
+            var user = new User(requestDto.username(), "fakeHashPassword");
+            return userRepository.save(user);
+        } else {
+            return optionalUser.get();
```

Comment:
Das müsste sich auf 
```
return optionalUser
    .orElse(userRepository.save(new User(requestDto.username(), "fakeHashPassword"))
``` 
kürzen lassen. Wenn dir das zu lange erscheint, könnte man den `orElse` Teil natürlich noch in eine extra Methode auslagern.